ios/Exponent/Kernel/Services/Notifications/EXApiV2Client+EXRemoteNotifications.m

```diff
@@ -39,17 +39,28 @@ - (NSURLSessionTask *)updateDeviceToken:(NSData *)deviceToken completionHandler:
 }
 
 
-- (NSURLSessionTask *)getExpoPushTokenForExperience:(NSString *)experienceStableLegacyId
+- (NSURLSessionTask *)getExpoPushTokenForEASProject:(nullable NSString *)easProjectId
+                           experienceStableLegacyId:(nullable NSString *)experienceStableLegacyId
                                         deviceToken:(NSData *)deviceToken
                                   completionHandler:(void (^)(NSString * _Nullable, NSError * _Nullable))handler
 {
   NSMutableDictionary *arguments = [NSMutableDictionary dictionaryWithDictionary:@{
     @"deviceId": [EXKernel deviceInstallationUUID],
-    @"experienceId": experienceStableLegacyId,
     @"appId": NSBundle.mainBundle.bundleIdentifier,
     @"deviceToken": deviceToken.apnsTokenString,
     @"type": @"apns",
   }];
+  
+  if (easProjectId != nil) {
+    arguments[@"projectId"] = easProjectId;
+  } else if (experienceStableLegacyId != nil) {
+    arguments[@"experienceId"] = experienceStableLegacyId;
+  } else {
+    @throw [NSException exceptionWithName:NSInvalidArgumentException
+                                   reason:@"Must supply either experienceId or projectId."
+                                 userInfo:@{}];
```

Comment:
I think it would be better to create an NSError and pass it to the completionHandler block. I traced backwards through the code and found that the completion handler passes errors to a `reject` block that sends an error back to JavaScript.

Even though this specific error can't be triggered from JS (AFAICT) and would be caused by a programming error or a configuration error (manifest w/out EAS project ID and stable legacy ID?), it's a nicer developer experience for the app not to crash and instead to get a red box with a readable error message.

Reply:
Yeah, it should. I was thrown off by the return type of the function (needing to still return the NSURLSessionTask) but I think we can actually convert it to void which will make calling the completionHandler here clean.